### PR TITLE
Add missing `news` entries and fix syntax in some of them

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Some of the merged PRs did not have `news` items because we hadn't caught up with the best practices by then. Some of those did have `news` files, but using markdown syntax instead of RST (main difference, how to add `monospace text`; i.e. single vs double backticks).

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by kenodegard** [comment]
@jaimergp I suggest we convert all RST to MD instead, this is what we've been doing in conda & conda-build

**Comment by jaimergp** [comment]
@kenodegard Should I convert the existing `CHANGELOG.rst` too?

**Comment by kenodegard** [comment]
@jaimergp yeah you'll also need to make some minor updates to the rever config, see https://github.com/conda/conda-build/pull/4483/commits/6bc25ecc0f9473d465a5d540936312d70dd5761a https://github.com/conda/conda/pull/11478/commits/cf5bb34646f9c202422ec2509b0bf706cceea79d

File: news/500-extra-files [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Enhancements:
+-------------
```
Comment (kenodegard) [inline news/500-extra-files:2]: This format may cause a problem when running rever?

Comment (jaimergp) [reply news/500-extra-files:2]: Oops, yep, totally missed that. Thanks!

**Review by jezdez — APPROVED** [review]

**Pull request merged**